packages/compass/src/main/auto-update-manager.ts

```diff
@@ -449,38 +445,77 @@ class CompassAutoUpdateManager {
     }
   }
 
+  static async downloadUpdate() {
+    try {
+      return new Promise<PromiseSettledResult<void>>((resolve) => {
+        autoUpdater.once('error', (reason) => {
+          resolve({ status: 'rejected', reason });
+        });
+
+        autoUpdater.once('update-downloaded', () => {
+          resolve({ status: 'fulfilled', value: undefined });
+        });
+      });
+    } finally {
+      // checkForUpdate downloads and installs the update when available, there
+      // is also no way to interrupt this process so once it starts, disabling
+      // updates in the options will not do anything
+      autoUpdater.setFeedURL(this.getFeedURLOptions());
+      autoUpdater.checkForUpdates();
+    }
+  }
+
   private static currentActionAbortController: AbortController =
     new AbortController();
 
-  static setState(newState: AutoUpdateManagerState, ...args: unknown[]) {
-    const currentStateHandlers = STATE_UPDATE[this.state];
+  static stop() {
+    this.currentActionAbortController.abort();
+  }
 
-    if (!currentStateHandlers) {
-      debug(`State ${this.state} doesn't support any state transitions`);
+  static async setState(newState: AutoUpdateManagerState, ...args: unknown[]) {
+    // Something already aborted the state transition outside of the setState
+    // loop. This might happen if exit hanler calling stop on the auto update
```

Comment:
```suggestion
    // loop. This might happen if exit handler calling stop on the auto update
```

but I’m honestly a bit unsure about how to parse the sentence as a whole here, I’m guessing either “calling” or “was called” should be removed?

Reply:
I am re-reading it and I've no idea how this sentence happened 😰 I'll rewrite it, thanks for spotting